### ISSUE TITLE
Message id is not a string

### ISSUE THREAD

**Issue by harry-73** [jamauro/pub-sub#7]
Hi,

I use:

- Meteor 3 beta 7
- jorgenvatle:vite-bundler@2.0.0-beta.14

Currently, I am not able to replicate in a small other project.

I have an issue using the package when I call a method:

```
W20240404-12:11:08.962(2)? (STDERR) meteor://💻app/packages/ddp-common.js:232
W20240404-12:11:08.963(2)? (STDERR)     throw new Error("Message id is not a string");
W20240404-12:11:08.963(2)? (STDERR)           ^
W20240404-12:11:08.963(2)? (STDERR) 
W20240404-12:11:08.964(2)? (STDERR) Error: Message id is not a string
W20240404-12:11:08.964(2)? (STDERR)     at DDPCommon.stringifyDDP (packages/ddp-common/utils.js:113:11)
W20240404-12:11:08.964(2)? (STDERR)     at Object.DDPCommon.stringifyDDP (packages/montiapm:agent/lib/hijack/wrap_ddp_stringify.js:7:21)
W20240404-12:11:08.964(2)? (STDERR)     at packages/ddp-server/livedata_server.js:563:36
W20240404-12:11:08.964(2)? (STDERR)     at packages/ddp-server/livedata_server.js:404:46
W20240404-12:11:08.964(2)? (STDERR)     at <anonymous>
W20240404-12:11:08.965(2)? (STDERR)     at processTicksAndRejections (node:internal/process/task_queues:95:5)
W20240404-12:11:08.965(2)? (STDERR) 
W20240404-12:11:08.965(2)? (STDERR) Node.js v20.11.1
W20240404-12:11:09.411(2)? (STDERR) ⚡  Meteor parent process is no longer running. Shutting down...
```

Client side call:
```
Meteor.callAsync('myMethod', { Id: myId })
          .then(() => {
```

Server side (the behavior is the same with Meteor.Method).
```
export const myMethod = new ValidatedMethod({
  name: 'myMethod',
  rateLimit: {
    numRequests: 5,
    timeInterval: 5000
  },
  validate: new SimpleSchema({
    Id: { type: Mongo.Collection.ObjectID },
    'Id._str': { type: String }
  }).validator(),

  run: async function ({ Id }) {
 
    await Notifications.insertAsync({ 
      ...    });

...
}
});
```

Thanks.

**Comment by harry-73** [comment]
I found some clues.

My code is the following:
```

const Markers = new Mongo.Collection('Markers', {
  idGeneration: 'MONGO'
});

Meteor.publish('markers', function () {
 
  if (!this.userId) return this.ready();

  return Markers.find({});
});
```

on Client Side:

`subscribe('markers');
`

With this code, I have the issue but I rename the publication from `markers` to `Markers`, it works well.

It seems, it is something in the `mongo.js` file around:

```
const shouldSendMutationUpdates = collectionName => {
  return (
    Meteor.settings.public.packages?.['jam:pub-sub']?.pubs.some(s => s.includes(collectionName))
    || !publishHandlers.some(s => s.includes(collectionName))
  )
}
```

**Comment by jamauro** [comment]
Thanks for the additional context. At the moment, collection names are assumed to follow the popular convention of being lowerCamelCase. So if you were to change to the following, it would work:

```js
const Markers = new Mongo.Collection('markers', {
  idGeneration: 'MONGO'
});
```

Is it important for you to maintain `Markers` as the collection name inside Mongo.Collection?

**Comment by harry-73** [comment]
OK. Thanks.  
This means database migration and a few procedures for updating production. This should be manageable.
In the meantime, I will clone it locally and do some modifications to support my setup.

**Comment by harry-73** [comment]
I will do something like that:

```
let publishHandlers = [];
Meteor.startup(() => {
  let OriginalPublishHandlers = Object.keys(Meteor.server.publish_handlers);
  publishHandlers = OriginalPublishHandlers.map(name => name.toLowerCase());
});

const shouldSendMutationUpdates = collectionName => {
  return (
    Meteor.settings.public.packages?.['jam:pub-sub']?.pubs.some(s => s.includes(collectionName))
    || !publishHandlers.some(s => s.includes(collectionName.toLowerCase()))
  )
}
```

**Comment by jamauro** [comment]
I just published a fix in `v0.2.3`. It allows you to maintain `Markers` as your collection name if you'd like and includes `Mongo.ObjectID` support if you choose to use `Meteor.publish.once` or `Meteor.publish.stream`. 

I think we can close this issue once you confirm you no longer see `Message id is not a string`.

**Comment by harry-73** [comment]
Thanks. It works.

I am closing the issue.
